### ISSUE TITLE
Error when receiving notification with deep link

### ISSUE THREAD

**Issue by JoakimMellonn** [aws-amplify/amplify-flutter#3547]
### Description

I'm trying to send a notification which opens a deep link, but it displays the error below. If I open the notification the app will crash with an uncaught error.

```shell
════════ Exception caught by services library ══════════════════════════════════
The following _TypeError was thrown during a platform message callback:
type '_Map<Object?, Object?>' is not a subtype of type 'Map<String, Object?>?' in type cast

When the exception was thrown, this was the stack
#0      new PushNotificationMessage.fromJson
push_notification_message.dart:24
#1      AmplifyPushNotificationsFlutterApi.onNotificationReceivedInBackground
push_notifications_flutter_api.dart:117
#2      PushNotificationsFlutterApi.setup.<anonymous closure>
native_push_notifications_plugin.g.dart:128
#3      BasicMessageChannel.setMessageHandler.<anonymous closure>
platform_channel.dart:212
#4      _DefaultBinaryMessenger.setMessageHandler.<anonymous closure>
binding.dart:479
#5      _invoke2 (dart:ui/hooks.dart:174:13)
#6      _ChannelCallbackRecord.invoke (dart:ui/channel_buffers.dart:40:5)
#7      _Channel.push (dart:ui/channel_buffers.dart:130:31)
#8      ChannelBuffers.push (dart:ui/channel_buffers.dart:326:17)
#9      PlatformDispatcher._dispatchPlatformMessage (dart:ui/platform_dispatcher.dart:686:22)
#10     _dispatchPlatformMessage (dart:ui/hooks.dart:86:31)
════════════════════════════════════════════════════════════════════════════════
``` 

This is the uncaught error, when opening the notification:
```shell
_TypeError (type '_Map<Object?, Object?>' is not a subtype of type 'Map<String, Object?>?' in type cast)
```

### Categories

- [ ] Analytics
- [ ] API (REST)
- [ ] API (GraphQL)
- [ ] Auth
- [ ] Authenticator
- [ ] DataStore
- [X] Notifications (Push)
- [ ] Storage

### Steps to Reproduce

1. Create flutter app with notifications enabled.
2. Send notification to device which opens a deep link.
3. Open the notification.

### Platforms

- [X] iOS
- [X] Android
- [ ] Web
- [ ] macOS
- [ ] Windows
- [ ] Linux

### Flutter Version

3.10.6

### Amplify Flutter Version

1.3.1

### Deployment Method

Amplify CLI

**Comment by dnys1** [comment]
Hi @JoakimMellonn, sorry you are facing this issue.

I've opened https://github.com/aws-amplify/amplify-flutter/pull/3557 to fix this.

**Comment by JoakimMellonn** [comment]
> Hi @JoakimMellonn, sorry you are facing this issue.
> 
> I've opened https://github.com/aws-amplify/amplify-flutter/pull/3557 to fix this.

Thank you! Do you know when the update will come out?

**Comment by dnys1** [comment]
I will work to get it out this week.

**Comment by JoakimMellonn** [comment]
Thank you @dnys1!

**Comment by dnys1** [comment]
The fix for this has been released in v1.3.3. Thanks for your patience, and please let us know if you continue to face any issues.
